server/routerlicious/packages/services-shared/src/storage.ts

```diff
@@ -129,7 +131,8 @@ export class DocumentStorage implements IDocumentStorage {
 		values: [string, ICommittedProposal][],
 		enableDiscovery: boolean = false,
 	): Promise<IDocumentDetails> {
-		const gitManager = await this.tenantManager.getTenantGitManager(tenantId, documentId);
+		const storageName = await this.storageNameProvider.createStorageName(tenantId, documentId);
```

Comment:
We could also introduce a new setting, say, `enableComputeStorageName`, and only take action when it is enabled

Reply:
Just curious if it is absolutely necessary to calculate the storageName here in CreateDocument flow? Ideally, we would defer this to the component (Historian) that directly interacts with backing storage (Azure Storage in case of FRS) to minimize the impact on other components and unnecessary customization. CreateDocument, largely, does not care where eventually the document data gets stored.

Reply:
@arikt-ms, @znewton , @hedasilv 
One thing I can think is to leverage the `createStorageName`/`assignStorageName` to historian, and historian do the `create/assign` logic in its component, and return the result to create doc flow.

I don't think we should reverse the sequence of create initial summary then create document in DB. The reason is only the initial summary upload create the folder if not exists, other component assume the folder already exists and don't create this folder on the fly. If we flip the sequence and say we had some issue of initial upload, the storage folder will not exist and a LOT logic might broken. I think if leave the document metadata at the end the is the right way.

Reply:
Change to no calculate storage name when create doc, but historian do that and report the name back to documents